In [1]:
import numpy as np

In [2]:
#comparison of hmagnitude for Saturn and Jupiter

#h=sqrt(mu*a*(a-e^^2)), but assuming circular obits, so e is zero, and mu is GM

msun=1.9981e30
mjupiter=1.898e27/msun
msaturn=5.683e26/msun
msun=msun/msun
ejupiter=0.0484
esaturn=0.0555 
ajupiter=5.2102 #AU
asaturn=9.5380 #AU


def h(a,e):  #program that returns the non-normalized specific angular momentum, multiply all 
    #the hx,hy,hz components to obtain full description
    return np.sqrt(a*(1-(e**2)))


hjupiter=h(ajupiter,ejupiter)
hsaturn=h(asaturn,esaturn)


Using what we know about centers of mass, we can also prove that the Sun does not contribute significantly to the invariant plane.

If we consider the sun at the origin, the center of mass between Jupiter and the Sun will be located at (in AU):


In [3]:

rcm=(msun*0+mjupiter*ajupiter)/(mjupiter+msun)  
#basically how far the Sun will be from the CM due to Jupiter, and Saturn's will be even less!

print(h(rcm,0))

0.0703170302593


So by assuming that in the Sun-Jupiter system the Sun orbits around the common center of mass in a circular orbit, this is the strongest possible contribution that the Sun could contribute towards the angular momentum. Calculating this value we obtain a value of 0.0703, which is over 32 times smaller than the contributions due to Jupiter's orbit, so it is safe to neglect this for the rest of the question.

In [4]:





iJ=np.deg2rad(1.3032)
oJ=np.deg2rad(100.4644)  #using values from Danby Appendix C 
hxJ=np.sin(oJ)*np.sin(iJ)*hjupiter  #this is currently a unit vector, so multiply by the magnitude solved for above
hyJ=-np.cos(oJ)*np.sin(iJ)*hjupiter
hzJ=np.cos(iJ)*hjupiter

LxJ=hxJ*mjupiter
LyJ=hyJ*mjupiter
LzJ=hzJ*mjupiter
#Saturn
iS=np.deg2rad(2.4888)
oS=np.deg2rad(113.6655)  #using values from Danby Appendix C 
hxS=np.sin(oS)*np.sin(iS)*hsaturn
hyS=-np.cos(oS)*np.sin(iS)*hsaturn
hzS=np.cos(iS)*hsaturn

LxS=hxS*msaturn
LyS=hyS*msaturn
LzS=hzS*msaturn


In [5]:


Lx=LxJ+LxS
Ly=LyJ+LyS
Lz=LzJ+LzS


In [6]:
print(msaturn/mjupiter)

0.2994204425711275


In [7]:
L=np.sqrt(pow(Lx,2)+pow(Ly,2)+pow(Lz,2))
print("before normalizing, L is equal to " + str(L))

before normalizing, L is equal to 0.00304258272648


In [8]:
#now to normalize. 
Lx=Lx/L
Ly=Ly/L
Lz=Lz/L

L=np.sqrt(pow(Lx,2)+pow(Ly,2)+pow(Lz,2))
print(L)

1.0


In [9]:
#solving for inclination
i=np.arccos(Lz)
print("inclination = " +str(np.rad2deg(i)) + " degrees")

inclination = 1.63421249707 degrees


In [10]:
#solving for omega, there are multiple equations to do this! All give the answer below: 
np.rad2deg(np.arcsin(Lx/np.sin(i)))-180    #gives a weird answer, explained below
np.rad2deg(np.arccos(-Ly/np.sin(i)))  #gives answer
omega=np.rad2deg(np.arctan2(Lx,-Ly))  #gives answer
print("omega = " +str(omega) + " degrees")

omega = 106.217094977 degrees


Due to the properties of feeding arcsin a positive number, it will always return a positive number (provided it is within the acceptable range). Because of this, using the equation hx=sin(i)sin(omega) will produce the -omega+180, once adjusted 



Comparing with the answers for the inclination and longitude of the ascending node in the textbook, these values are very close!

In [11]:
#According to section 9.1,

iactual=1+35/60
omegaactual=107

#So jupiter and saturn make up for 

print(np.rad2deg(i)/iactual)
print(omega/omegaactual)



1.03213420868
0.992683130622


Or nearly all the eccentricity, and the same goes for the longitude of the ascending node are due to the contributions of Jupiter and Saturn.